main.nf

```diff
@@ -345,66 +461,102 @@ process bowtie_miRBase_hairpin {
     file index from hairpin_index
 
     output:
-    file '*.hairpin.bam' into miRBase_hairpin_bam
+    file '*.hairpin.bam' into miRBase_hairpin_bam, miRBase_hairpin_bam_mirtop
     file '*.hairpin_unmapped.fq.gz' into hairpin_unmapped_reads
 
     script:
     index_base = index.toString().tokenize(' ')[0].tokenize('.')[0]
     prefix = reads.toString() - '.mature_unmapped.fq.gz'
+    seq_center = params.seq_center ? "--sam-RG ID:${prefix} --sam-RG 'CN:${params.seq_center}'" : ''
     """
     bowtie \\
         $index_base \\
-        -p 2 \\
+        -p ${task.cpus} \\
         -t \\
-        -k 1 \\
-        -m 1 \\
+        -a \\
         --best \\
         --strata \\
         -e 99999 \\
         --chunkmbs 2048 \\
         -q <(zcat $reads) \\
         --un ${prefix}.hairpin_unmapped.fq \\
-        -S \\
+        -S $seq_center \\
         | samtools view -bS - > ${prefix}.hairpin.bam
 
     gzip ${prefix}.hairpin_unmapped.fq
     """
 }
 
+/*
+ * STEP 4.1 - Bowtie against miRBase hairpin with collapsed reads
+ */
+process bowtie_miRBase_hairpin_collapsed {
+    label 'process_medium'
+    tag "$reads"
+    
+    input:
+    file reads from collapsed_fasta
+    file index from hairpin_index
+
+    output:
+    file '*.bam' into miRBase_hairpin_collapse_bam
+
+    script:
+    index_base = index.toString().tokenize(' ')[0].tokenize('.')[0]
+    prefix = reads.baseName
+    seq_center = params.seq_center ? "--sam-RG ID:${prefix} --sam-RG 'CN:${params.seq_center}'" : ''
+    """
+    bowtie \\
+        $index_base \\
+        -p ${task.cpus} \\
+        -t \\
+        -k 50 \\
+        -a \\
+        --best \\
+        --strata \\
+        -e 99999 \\
```

Comment:
Are these parameters that we should make configurable for users?

Reply:
I don't think this should be configurable. It could mess up really the accuracy of the pipeline.